### DISCUSSION TITLE
next config file now has .mjs instead of .js?

### DISCUSSION THREAD

**Discussion by BearCooder** [vercel/next.js#60945] [Help] [answered]
### Summary

Hi, I noticed that the recent next version after installing from scratch, has next.config.mjs instead of next.config.js?
What is the reason for this change? 

I noticed this due to the error `ReferenceError: require is not defined in ES module scope, you can use import instead`
This is the default config I use in order to use next-intl library:

```
/** @type {import('next').NextConfig} */
const withNextIntl = require('next-intl/plugin')('./i18n/i18n.ts');

module.exports = withNextIntl({});
```
Changing from .mjs to .js solves this problem. But I am curious why the change?

**Comment by joulev** [comment, marked as answer]
`require` and `module.exports` are invalid in ES modules (`.mjs` files). If you do use `.mjs` you need to use module import and export statements. For your config file that would be

```js
import nextIntl from 'next-intl/plugin';

const withNextIntl = nextIntl('./i18n/i18n.ts');

export default withNextIntl({});
```

**Reply by Farhan160200** [reply]
How do i change my next.config.js that include 

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    appDir: true,
    serverComponentsExternalPackages: ["mongoose"],
  },
  images: {
    domains: ['lh3.googleusercontent.com'],
  },
  webpack(config) {
    config.experiments = {
      ...config.experiments,
      topLevelAwait: true,
    }
    return config
  }
}
module.exports = nextConfig
```

**Could you make it to the .mjs file format**

**Reply by joulev** [reply]
@Farhan160200 Why do you want to change to `.mjs`? `.js` is still working perfectly fine. `.mjs` is preferred but that doesn't mean `.js` stops working. For your code it doesn't have anything that requires you to switch to `.mjs`.

**Reply by Farhan160200** [reply]
**cause basicly when i installed next.js the next.config file already in .mjs format. So when i change file to .js format it appears error** 

```
⚠ Invalid next.config.js options detected:
⚠ Unrecognized key(s) in object: 'appDir' at "experimental"
⚠ See more info here: https://nextjs.org/docs/messages/invalid-next-config

```
So i wanna change my previous code to .mjs format

**Reply by joulev** [reply]
@Farhan160200 The cause of the error is not due to the `.mjs` file format, it's due to `experimental.appDir` which has been removed. Remove the `appDir: true` from your `experimental` object.

**Reply by Farhan160200** [reply]
Wokrss, thanks @joulev

**Reply by zacBkh** [reply]
Hello!
Thanks a lot but I have personnally two things to export from this config file, internationalization (withNextIntl) but also something for PWA support (serwist)
I don't manage to find the syntax

**Reply by joulev** [reply]
@zacBkh This maybe, I have never used either of them before

```tsx
import nextIntl from 'next-intl/plugin';
import withSerwistInit from "@serwist/next";

const withNextIntl = nextIntl('./i18n/i18n.ts');
const withSerwist = withSerwistInit({
    swSrc: "app/sw.ts",
    swDest: "public/sw.js",
});

export default withSerwist(withNextIntl({}));
```

**Reply by ISHA50** [reply]
Hey, can you help me? I'm building documentation using Nextra. I fixed the previous error as you suggested, but now I'm encountering another error. I'm not sure how to resolve it .   
[Error.pdf](https://github.com/vercel/next.js/files/14968881/Error.pdf)

**Reply by joulev** [reply]
@ISHA50 That error doesn’t look related to the main topic in this thread, but rather related to Nextra itself. Please seek help in a Nextra support community.

**Reply by ayushghatkar8080** [reply]
import { withSentryConfig } from "@sentry/nextjs";
/* @type {import('next').NextConfig} */
import { withNextVideo } from "next-video/process";

const nextConfig = {
  images: {
    remotePatterns: [
      {
        hostname: "zyx.asdsad.com",
      },
    ],
  },
};

export default withSentryConfig(withNextVideo(nextConfig),
  {
    silent: true,
    org: "org-name",
    project: "pro-name",
  },
  {
    widenClientFileUpload: true,
    transpileClientSDK: true,
    hideSourceMaps: true,
    disableLogger: true,
    automaticVercelMonitors: true,
  }
);



hostname "zyx.asdsad.com" is not configured under images in your `next.config.js`

**Comment by srunrotha604** [comment]
Is anyone able to help me set the customer header title in Nextjs 14?  when i change the page.  when i check in nextjs document they provide to change in next config file but i still see localhost as my header tittle, can you help me with this ? 
![Uploading Screenshot 2024-05-23 at 11.14.40 AM.png…]()

**Comment by keithmanilla** [comment]
Just change `module.exports = nextConfig` into `export default nextConfig;`:
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  output: 'export',
}

export default nextConfig;
```
